packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3089,6 +3094,8 @@ export class ContainerRuntime
 					if (this.dirtyContainer !== checkpointDirtyState) {
 						this.updateDocumentDirtyState(checkpointDirtyState);
 					}
+					stageControls?.discardChanges();
```

Comment:
I'm having trouble rationalizing what happens if you start calling `orderSequentially` across async frames.  As soon as the Outbox flushes, the `checkpoint` becomes invalid I think.

Reply:
orderSequentially is always synchronous, so i don't think that can happen

Reply:
Actually, we'll assert as soon as flush happens (which will be real soon after going async, see `scheduleFlush`)

Reply:
> orderSequentially is always synchronous, so i don't think that can happen

I think I'm getting ahead of myself - I thought maybe you were trying to change that.